packages/iov-cosmos/src/cosmosconnection.ts

```diff
@@ -75,34 +167,79 @@ export class CosmosConnection implements BlockchainConnection {
   public async getTx(
     id: TransactionId,
   ): Promise<(ConfirmedAndSignedTransaction<UnsignedTransaction>) | FailedTransaction> {
-    throw new Error("not implemented");
+    try {
+      const response = await this.restClient.txsById(id);
+      const chainId = await this.chainId();
+      return this.parseAndPopulateTxResponse(response, chainId);
+    } catch (error) {
+      if (error.response.status === 404) {
+        throw new Error("Transaction does not exist");
+      }
+      throw error;
+    }
   }
 
   public async postTx(tx: PostableBytes): Promise<PostTxResponse> {
-    throw new Error("not implemented");
+    const { txhash, raw_log } = await this.restClient.postTx(tx);
+    const transactionId = txhash as TransactionId;
+    const firstEvent: BlockInfo = { state: TransactionState.Pending };
+    const producer = new DefaultValueProducer<BlockInfo>(firstEvent);
+    return {
+      blockInfo: new ValueAndUpdates<BlockInfo>(producer),
+      transactionId: transactionId,
+      log: raw_log,
+    };
   }
 
   public async searchTx(
     query: TransactionQuery,
   ): Promise<readonly (ConfirmedTransaction<LightTransaction> | FailedTransaction)[]> {
-    throw new Error("not implemented");
+    const queryString = buildQueryString(query);
+    const chainId = await this.chainId();
+    const { txs: responses } = await this.restClient.txs(queryString);
+    return Promise.all(responses.map(response => this.parseAndPopulateTxResponse(response, chainId)));
   }
 
   public listenTx(
-    query: TransactionQuery,
+    _query: TransactionQuery,
   ): Stream<ConfirmedTransaction<LightTransaction> | FailedTransaction> {
     throw new Error("not implemented");
   }
 
-  public liveTx(query: TransactionQuery): Stream<ConfirmedTransaction<LightTransaction> | FailedTransaction> {
+  public liveTx(
+    _query: TransactionQuery,
+  ): Stream<ConfirmedTransaction<LightTransaction> | FailedTransaction> {
     throw new Error("not implemented");
   }
 
   public async getFeeQuote(tx: UnsignedTransaction): Promise<Fee> {
-    throw new Error("not implemented");
+    if (!isSendTransaction(tx)) {
+      throw new Error("Received transaction of unsupported kind.");
+    }
+    return {
+      tokens: {
+        fractionalDigits: 6,
+        quantity: "5000",
+        tokenTicker: atom,
+      },
+      gasLimit: "200000",
+    };
```

Comment:
Do we have any documentation why those values are hardcoded?

Reply:
I can't actually remember where I got these from. I see it in an example here: https://github.com/cosmostation/cosmosjs#usage

I don't know what counts as a sensible default, and taking a brief look at a block explorer it seems like there are wildly varying values used on mainnet.

Reply:
👍 we can adapt this on demand. The worst that can happen is overpay or tx rejections.